internal/weldr/api.go

```diff
@@ -1411,8 +1411,25 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
 			statusResponseError(writer, http.StatusBadRequest, errors)
 			return
 		}
+		targets = append(targets, t)
 	}
 
+	filename, _, err := api.distro.FilenameFromType(cr.ComposeType)
+	if err != nil {
+		errors := responseError{
+			ID:  "BadCompose",
+			Msg: fmt.Sprintf("internal error: no filename for compose type %s", cr.ComposeType),
+		}
+		statusResponseError(writer, http.StatusBadRequest, errors)
+		return
+	}
+
+	targets = append(targets, target.NewLocalTarget(
```

Comment:
Before this patch, this was skipped if the store did not have a state directory. I think that's the wrong place to have a conditional (do it when receiving the file, if anything), so this should be ok (that was only conditional for easier testing, so might never matter at all). But worth a comment in the commit message?

Reply:
Good point. Added an explanation.